### ISSUE TITLE
Variables in @import statement cause for unexpected result

### ISSUE THREAD

**Issue by mantsevich** [less/less.js#3374]
Variables in @import statement cause for unexpected result.

I prepared demo app.
https://github.com/mantsevich/less-variables-import-issue-demo 

1) download
2) npm i
3) npm start
4) compare "correct.css" & "issue.css" files.

What do we have:
- Two similar files: _entry-correct.less_ & _entry-issue.less_ import 2 components: _hello-galaxy_ & _hello-world_
- _hello-world_ imports _hello-galaxy_ component with (reference) flag

Difference in the source code: _entry-issue.less_ uses variable in **@import** statement.
Difference in the result: _issue.css_ doesn't include _hello-galaxy_ component at all.

**Expected result**: Result of CSS files should be identical.

**Comment by seven-phases-max** [comment]
Btw. neither `correct.css` nor `issue.css` does not quite match the sources so the project is a bit difficult to follow. Also your `@path-to-components` is `"components/"` but than it's used as `"./@{path-to-components}/..."` resulting in "./components//..." - I'd expect this to trigger a error (does it?).

---
Either way at quick glance I suspect it's about the "`once` import rule" and the order of imports:
Since `hello-world.less` already imports `hello-galaxy.less`, the second `hello-galaxy.less` import (the one in `entry.less`) is ignored. Thus different result with and without `(reference)` of *the only* applied import is expected.
In other words, it seems to be the same issue as #3040 (i.e. current behavior is expected, but there's room for a feature request *if* someone is interested to design/implement that).

**Comment by mantsevich** [comment]
I locally remove extra "/" - result the same.

-------------

I think it's different issue. 
Why does code work if I don't use variable in the path?

Working:
```less
@import "./components/hello-world/hello-world.less";
@import "./components/hello-galaxy/hello-galaxy.less";
```

With issue:
```less
@import "./@{path-to-components}/hello-world/hello-world.less";
@import "./@{path-to-components}/hello-galaxy/hello-galaxy.less";
```

**Result**: Correct entry - exports _hello-galaxy_, but issued (with variables in the path) - don't.

**Comment by seven-phases-max** [comment]
I tested this in both Less `3.9.0` and `2.7.3` and the issue that inconsistency being wider than just "variable-in-import-vs-reference".

---

Here's a few **minimalistic examples** to illustrate the inconsistency:
*whatever.less:*
```less
whatever {whatever: whatever}
```

*indirect.less:*
```less
@import "whatever.less";
```

*indirect-ref.less:*
```less
@import (reference) "whatever.less";
```

---
*main-1:*
```less
@import (reference) "whatever.less";
@import "whatever.less";    // not imported

/* something */
```

*main-2:*
```less
@import "indirect-ref.less";
@import "whatever.less";    // imported

/* something */
```

*main-3:*
```less
@w: w;
@import "indirect-ref.less";
@import "@{w}hatever.less"; // not imported

/* something */
```

*main-4:*
```less
@import "indirect-ref.less";
@import "indirect.less";    // not imported

/* something */
```


Labeling as a bug since regardless of what result we expect, what is important is that we expect the same output from all these `main` files (there're obviously other (in)direct-import/reference-vs-ones/var-interp combinations to imagine to suffer from this).

---

I guess this is related to https://github.com/less/less.js/pull/2729 (i.e. most likely a `main-2`-like code triggers some importing-chain patterns introduced there), thus it's basically a continuation of the issues mentioned in the PR.
Though I was not able to test Less `2.5.3` (the version before the #2729 patch) to make sure, as it no longer works with recent Node.js versions.

---
@mantsevich Still notice that technically it's `entry-less.less` producing "wrong" result if compared to other snippets (despite it making more logical sense for your use-case). And the variable in the import only triggers the "proper" ("currently expected" per #3040) behavior I described in my previous post.
